test/integration/acceptance/set_account_detail_test.cpp

```diff
@@ -66,6 +66,44 @@ TEST_F(SetAccountDetail, Self) {
       .done();
 }
 
+/**
+ * @given a user with can_set_detail perm
+ *        AND a json-object value with single quotes
+ * @when execute tx with SetAccountDetail command aimed to the user
+ * @then there is the tx in proposal
+ */
+TEST_F(SetAccountDetail, JsonObject) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(complete(baseTx(kUserId, kKey, "{'value' : ['abc', 'dca']}")))
```

Comment:
I suggest a bit stronger verification for both test cases:

Set the tested value as const at the beginning of a test case.
SendTx(Set the value as an account detail) - already done.
Query the detail.
ASSERT_EQ(const_detail, query_response_detail);